code/mob/living/silicon/robot.dm

```diff
@@ -2891,6 +2891,20 @@
 	proc/compborg_take_critter_damage(var/zone = null, var/brute = 0, var/burn = 0)
 		TakeDamage(pick(get_valid_target_zones()), brute, burn)
 
+	proc/collapse_to_pieces()
+		src.visible_message("<span class='alert'><b>[src]</b> falls apart into a pile of components!</span>")
+		var/turf/T = get_turf(src)
+		for(var/obj/item/parts/robot_parts/R in src.contents)
+			R.set_loc(T)
```

Comment:
I'm unfamiliar with robot code. Would 100% damaged (utterly destroyed) limbs still be in contents?

Reply:
For that matter, do drop limbs store damage? (pretty sure not) However breaking down a robot in order to repair it magically seems an odd edge case we might not need to worry about.

Reply:
> I'm unfamiliar with robot code. Would 100% damaged (utterly destroyed) limbs still be in contents?

They would not!